### DISCUSSION TITLE
Lyhyitä koodipätkiä

### DISCUSSION THREAD

**Discussion by villevuor** [partio-scout/kipa#26] [Show and tell]
Varmaan jokainen kisa rakentaa omia lyhyitä skriptejä ym. Kipan täydennykseksi. Niitä voi jakaa tähän ketjuun, niin muutkin pääsee hyödyntämään, vaikka ei olisikaan niin kokenut koodari.

**Comment by villevuor** [comment]
**Yksinkertaiset varmuuskopiot Kipasta erilliselle koneelle**

```
while true; do wget -O kipa_backup_$(date +%F_%H%M%S).xml http://192.168.1.9:8000/kipa/UKKO2018/tallenna/; sleep 1200; done
```

Aja komento terminaalissa ja jätä pyörimään. Hakee 20 min välein (1200 sek) kipan backup-tiedoston ja tallentaa sen. Korjaa osoite oikeaksi. Jos Kipaa pyörittävä kone hajoaa, backupit kisasta on tallessa koneella joka ajaa tätä skriptiä (ja ihan toki siltäkin varalta, että joku vahingossa poistaa jonkun sarjan/tehtävän tms.).

Jos kipa on http-autentikaation takana, ne voi lisätä wget:n parametreiksi tarvittaessa.

**Comment by ZeiP** [comment]
Yksi vähän pidempi skripti löytyy paketista mukana, eli quick'n'dirty-skripti tehtäväkohtaisten käyttäjätunnusten luomiseen Nginx:n basic-authia varten: https://github.com/partio-scout/kipa/blob/master/scripts/generate_passwords.py

**Comment by Pena86** [comment]
Jos käyttää versiota, jossa ei ole kipassa käyttäjillä käyttöoikeusmäärittelyjä, joskus olen nähnyt, että päävalikosta kommentoidaan pois linkki, jolla pääsee poistamaan kisan.
